rockcraft/extensions/go.py

```diff
@@ -206,8 +207,8 @@ def _assets_stage(self):
             ]
         return user_stage
 
-    def _get_nested(self, obj: dict, path: str) -> dict:
-        """Get a nested object using a path (a dot-separated list of keys)."""
-        for key in path.split("."):
+    def _get_nested(self, obj: dict, paths: list[str]) -> dict:
```

Comment:
I did mean `*paths: str` instead of `paths: list[str]` so that callers can do `self._get_nested(data, "a", "b", "c")` instead of `self._get_nested(data, ["a", "b", "c"])`, but I suppose it doesn't make much of a difference